packages/admin-graphql-api-utilities/src/index.ts

```diff
@@ -10,10 +27,60 @@ export function parseGid(gid: string): string {
   throw new Error(`Invalid gid: ${gid}`);
 }
 
-export function composeGid(key: string, id: number | string): string {
-  return `gid://shopify/${key}/${id}`;
+export function parseGidWithParams(gid: string): ParsedGID {
+  // appends forward slash to help identify invalid id
+  const id = `/${gid}`;
+  const matches = GID_REGEXP.exec(id);
```

Comment:
Same question here? This GID is like an URL, can we use URLSearchParams instead? I have no idea of the cost tho.

Reply:
If we're using URLSearchParams, then could we go even further by using [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL_API) (and then use [url.searchParams](https://nodejs.org/docs/latest-v8.x/api/url.html#url_url_searchparams)) to parse the whole thing instead of writing regexes to do the first bit of parsing?

What's the browser support for URL / URLSearchParams like? at first glance it looks like [it's in everything we support](https://caniuse.com/#search=url)

Reply:
There's a test for "unprefixed gid" and forcing a gid to really be a full gid with protocol will be a breaking change :/

Reply:
oof. I'm guessing something like `if (!gid.startsWith('gid://') { id = `gid://${gid}` } is just gonna make this more confusing :(